GNUmakefile

```diff
@@ -78,4 +78,15 @@ license-check:
 tools:
 	go generate -tags tools tools/tools.go
 
-.PHONY: build test testall testacc cassettes vet fmt fmtcheck errcheck test-compile tools get-test-deps license-check
+docs: tools
+	tfplugindocs
+	@#This resource's documentation isn't currently generated
+	@rm -rf docs/resources/slo_correction.md
+	@# Timeboard + Screenbaord is deprecated
+	@# There is an issue with the security_monitoring schema that requires the docs to be updated by hand if needed
+	@git checkout HEAD -- docs/resources/timeboard.md docs/resources/screenboard.md docs/data-sources/security_monitoring_rules.md docs/resources/security_monitoring_default_rule.md docs/resources/security_monitoring_rule.md
+
+check-docs: docs
+	@if [ "`git status --porcelain docs/`" ] ; then echo "Uncommitted changes were detected in the autogenerated docs folder. Please run 'make docs' to autogenerate the docs" && echo `git status --porcelain docs/` && exit 1; else echo "Success: No generated documentation changes detected"; fi
```

Comment:
Can we reformat this line for readability

Reply:
Done 🙂 Its on the verge of warranting its own script in the `scripts` folder, but I think for now its OK inline here.